bigquery_etl/shredder/delete.py

```diff
@@ -251,6 +251,7 @@ def create_job(client):
                   )
                 ON
                   {field} = _source_{index}
+                  AND {partition.on_condition}
```

Comment:
This initially struck me as odd, since the partition condition only references the target table and not the other side of the join, but I finally grokked that this is forcing the join to fail for target rows that don't meet the partition condition so that `_source_{index} IS NULL` is true and all such rows are still included.

Another way to do that would be `OR {partition.condition} IS NOT TRUE` in the where clause, which I'm guessing would be a bit more computationally efficient, but having the logic to switch between `AND`/`OR` for the different cases seems like it'd be super awkward, so what you've got here is probably better in terms of readability/maintainability.